Src/DependencyCollector/Shared/Implementation/DependencyTableStore.cs

```diff
@@ -29,8 +31,18 @@ internal static DependencyTableStore Instance
         {
            get 
            {
-              Interlocked.CompareExchange<DependencyTableStore>(ref instance, new DependencyTableStore(), null);
-              return instance;
+               if (instance == null)
+               {
+                   lock (SyncRoot)
+                   {
+                       if (instance == null)
```

Comment:
C# needs doubled checking, see https://msdn.microsoft.com/en-us/library/ff650316.aspx

I do not know why initially this was implemented as singleton. If you say it's not needed, I'm fine to initialize instance once in crot